cms/articles/tests/test_pages.py

```diff
@@ -405,3 +409,104 @@ def test_correct_template_used_for_edit_page(self):
+            stream_data=[
+                ("manual_link", manual_dataset),
+            ],
+        )
+        self.page.save_revision().publish()
+        response = self.client.get(self.page.url)
+        content = response.content.decode(encoding="utf-8")
+
+        self.assertIn("Explore Data", content)
+        self.assertIn("View data used in this article", content)
+
+    def test_related_data_link_not_present(self):
+        """Test that the related data link is not rendered when there is no related data for the article."""
+        response = self.client.get(self.page.url)
+        content = response.content.decode(encoding="utf-8")
+
+        self.assertNotIn("Explore Data", content)
+        self.assertNotIn("View data used in this article", content)
+
+    def test_related_data_page_single_page(self):
+        """Test that pagination is not shown when the content fits on a single page."""
+        manual_dataset = {"title": "test manual", "description": "manual description", "url": "https://example.com"}
+
+        self.page.datasets = StreamValue(
+            DatasetStoryBlock(),
+            stream_data=[
+                ("manual_link", manual_dataset),
+            ],
+        )
+        self.page.save_revision().publish()
+        response = self.client.get(self.page.url + "related-data/")
+        content = response.content.decode(encoding="utf-8")
+
+        self.assertNotIn('class="ons-pagination__item ons-pagination__item--previous"', content)
+        self.assertNotIn('class="ons-pagination__item ons-pagination__item--current"', content)
+        self.assertNotIn('class="ons-pagination__item ons-pagination__item--next"', content)
+
+    @override_settings(RELATED_DATASETS_PER_PAGE=1)
+    def test_related_data_page_with_pagination(self):
+        """Test that pagination is shown when there is more than one page."""
+        manual_datasets = [
+            {"title": "test1", "description": "test", "url": "https://example.com/1"},
+            {"title": "test2", "description": "test", "url": "https://example.com/2"},
+            {"title": "test3", "description": "test", "url": "https://example.com/3"},
+            {"title": "test4", "description": "test", "url": "https://example.com/4"},
+        ]
+        self.page.datasets = StreamValue(
+            DatasetStoryBlock(),
+            stream_data=[("manual_link", dataset) for dataset in manual_datasets],
+        )
+        self.page.save_revision().publish()
+        response = self.client.get(self.page.url + "related-data/?page=2")
+        content = response.content.decode(encoding="utf-8")
+
+        self.assertIn('class="ons-pagination__item ons-pagination__item--previous"', content)
+        self.assertIn('class="ons-pagination__item ons-pagination__item--current"', content)
+        self.assertIn('class="ons-pagination__item ons-pagination__item--next"', content)
+        self.assertIn('class="ons-pagination__position">Page 2 of 4', content)
```

Comment:
we need a test that confirms that adding a new article in the series copies over the previous one's datasets and sorting setting

Reply:
Added in [1a7a89b](https://github.com/ONSdigital/dis-wagtail/pull/165/commits/1a7a89be9bd8fb2dd583a492e5330ba3206d691c)